qiskit/wrapper/_wrapper.py

```diff
@@ -19,13 +19,16 @@
 _DEFAULT_PROVIDER = DefaultQISKitProvider()
 
 
-def register(token, url='https://quantumexperience.ng.bluemix.net/api',
-             hub=None, group=None, project=None, proxies=None, verify=True,
-             provider_name=None):
+def register(*args, provider_class=IBMQProvider, provider_name='ibmq', **kwargs):
     """
     Authenticate against an online backend provider.
 
     Args:
+        args (tuple): positional arguments passed to provider class initialization
+        provider_class (BaseProvider): provider class
+        provider_name (str): name for provider
```

Comment:
I would explain this a bit more, e.g. "optional user-assigned name for the provider"